### DISCUSSION TITLE
jsp:include & ios

### DISCUSSION THREAD

**Discussion by Osehwa** [niklasvh/html2canvas#3183] [Q&A]
The file profileMain.jsp contains html2canvas code, and the page I want to share using html2canvas is embedded within profileMain.jsp through the following tag: <jsp:include page="../today/todayMain_profile.jsp" />.

It works on Android but not on iOS. Can you explain why?

=====================================================
[Part of 'profileMain.jsp'  ]
```
		<section class="tabWrapper">
			<div id="todayTab" class="tabs focus">
				<div id="todayContent"></div> 			
					 <jsp:include page="../today/todayMain_profile.jsp" />
			</div>
		</section>

<script>
    		if(userAgent.match(/iPhone | iPad | iPod/)) {
    			console.log("ios");
    			console.log(shareArea);

    			html2canvas(shareArea).then((canvas) => {
        			console.log(canvas);
.
.
.
.
</script>
```
===========================================================

console.log(canvas) does not appear in the console.
console.log("ios") and  console.log(shareArea) appear in the console. (shareArea is HTML code)

**Comment by dcwang-lab** [comment]
2099961945 可以发我qq帮你看下

**Reply by Osehwa** [reply]
I'm sorry, but can you explain it in more detail? It hasn't been a year since I developed it, so I'm still a beginner
